localstack/services/awslambda/provider.py

```diff
@@ -385,16 +385,23 @@ def _create_version_model(
                 "The Revision Id provided does not match the latest Revision Id. Call the GetFunction/GetAlias API to retrieve the latest Revision Id",
                 Type="User",
             )
-        current_hash = current_latest_version.config.code.code_sha256
-        if (
-            code_sha256
-            and current_hash != code_sha256
-            and not current_latest_version.config.code.is_hot_reloading()
-        ):
+
+        # check if code hashes match if they are specified
+        current_hash = (
+            current_latest_version.config.code.code_sha256
+            if current_latest_version.config.package_type == PackageType.Zip
+            else current_latest_version.config.image.code_sha256
+        )
+        is_hot_reloading = (
+            current_latest_version.config.package_type == PackageType.Zip
```

Comment:
nit: the nested hot reloading check is a bit cumbersome but that's because it only applies to code package_type